src/common/assert.cc

```diff
@@ -40,8 +40,10 @@ namespace ceph {
     g_assert_line = line;
     g_assert_func = func;
     g_assert_thread = (unsigned long long)pthread_self();
+#if defined(HAVE_PTHREAD_GETNAME_NP)
```

Comment:
does this file `#include "include/compact.h"` indirectly? if yes, probably we can just call `ceph_pthread_getname()` here?